phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CompactionScanner.java

```diff
@@ -84,45 +110,99 @@ public CompactionScanner(RegionCoprocessorEnvironment env,
             Store store,
             InternalScanner storeScanner,
             long maxLookbackInMillis,
-            byte[] emptyCF,
-            byte[] emptyCQ,
-            int phoenixTTL,
-            boolean isSystemTable) {
+            PTable table) throws IOException {
         this.storeScanner = storeScanner;
         this.region = env.getRegion();
         this.store = store;
         this.env = env;
-        this.emptyCF = emptyCF;
-        this.emptyCQ = emptyCQ;
+        this.emptyCF = SchemaUtil.getEmptyColumnFamily(table);
+        this.emptyCQ = table.getEncodingScheme() == PTable.QualifierEncodingScheme.NON_ENCODED_QUALIFIERS ?
```

Comment:
Added SchemaUtil.getEncodingScheme(table), I am assuming that's what u meant

Reply:
I meant SchemaUtil.getEmptyColumnName(table) or maybe better to call it SchemaUtil.getEmptyColumnQualifier(table). Since we have SchemaUtil.getEmptyColumnFamily(table) and we should also have SchemaUtil.getEmptyColumnQualifier(table)